docker/Dockerfile

```diff
@@ -1,97 +1,53 @@
-MAINTAINER ivan.serezhkin@insolar.io
-
-WORKDIR /opt
-RUN set -x && \
-        cp -rf /go/src/github.com/insolar/insolar/keys /opt || true && \
-        /go/src/github.com/insolar/insolar/scripts/build/genesis/genesis.sh
-
-#                              GENESIS CONTAINER
-FROM alpine:3.9 AS genesis
-MAINTAINER ivan.serezhkin@insolar.io
-
-RUN mkdir -p /opt/genesis
-COPY --from=insolard-genesis-builder    \
-        [ "/opt/data.tgz",              \
-          "/opt/keys.tgz",              \
-          "/opt/genesis/" ]
-VOLUME [ "/opt/output" ]
-CMD ['cp', '/opt/genesis', '/opt/output/genesis']
-
-#                             INSGORUND CONTAINER
-FROM alpine:3.9 AS insgorund
-MAINTAINER ivan.serezhkin@insolar.io
-
-RUN set -x && \
-        mkdir -p /var/log /opt/bin /tmp/insgorund /opt/insgorund && \
-        apk add --no-cache --virtual /opt/.runtime-deps \
-            bash    \
-            curl
-
-COPY --from=insolard-genesis-builder /opt/bin/insgorund /opt/bin/insgorund
-COPY docker/insgorund-entrypoint.sh /opt/bin/docker-entrypoint.sh
+# Install required tools step (adds ~400mb)
+RUN echo -n "<DISK USAGE> before install_build_tools: " && du -sh / 2>/dev/null || true
+ADD ./scripts/build ./scripts/build
+RUN ./scripts/build/install_build_tools.sh
+
+# Install go dependencies step (adds ~800mb)
+RUN echo -n "<DISK USAGE> before vendor: " && du -sh / 2>/dev/null || true
+ADD Gopkg.* ./
+RUN dep ensure --vendor-only
+
+# adds sources (~15mb)
+RUN echo -n "<DISK USAGE> before insolar sources copy: " && du -sh / 2>/dev/null || true
+ADD . /go/src/github.com/insolar/insolar/
+
+RUN echo -n "<DISK USAGE> before build: " && du -sh / 2>/dev/null || true
+# pass build variables as arguments to avoid adding .git directory
+ARG BUILD_NUMBER
+ARG BUILD_DATE
+ARG BUILD_TIME
+ARG BUILD_HASH
+ARG BUILD_VERSION
+# build step: adds 400mb
+# not really cachebale because it uses BUILD_TIME variable
+# (we can mock BUILD_TIME, but it's not worth it, besacuse build time is around 1 min)
+RUN BUILD_NUMBER=${BUILD_NUMBER} \
+    BUILD_DATE=$BUILD_DATE \
+    BUILD_TIME=$BUILD_TIME \
+    BUILD_HASH=$BUILD_HASH \
+    BUILD_VERSION=$BUILD_VERSION \
+    make build
+RUN echo -n "<DISK USAGE> after build: " && du -sh / 2>/dev/null || true
+
+# 2) Base image for running test tests and binaries distribution iamges
+#    (approx local build time ~30s)
+FROM golang:1.12.9
+WORKDIR /go/src/github.com/insolar/insolar
 
-EXPOSE 7777 8002
+# adds 20mb
+RUN echo -n "<DISK USAGE> before apt-get: " && du -sh / 2>/dev/null || true
+# tools for functest (launchnet)
+RUN apt-get -y update && apt-get -y install jq lsof psmisc && apt-get clean
 
-ENTRYPOINT [ "/opt/bin/docker-entrypoint.sh" ]
-CMD []
+# adds 350mb (>250mb – binaries)
+RUN echo -n "<DISK USAGE> before insolar files copy: " && du -sh / 2>/dev/null || true
+COPY --from=build /go/src/github.com/insolar/insolar /go/src/github.com/insolar/insolar
+RUN ls -al ./bin && ls -al ./
```

Comment:
why?

Reply:
Just to be sure we not forget smth, could be useful in CI builds. I can remove/commenting it.